apps/backend/src/tests/response_api.test.ts

```diff
@@ -0,0 +1,555 @@
+  username: 'antonina',
+  password: 'secret',
+};
+
+const BOBS_LOGIN_DATA: LoginData = {
+  username: 'bob1234',
+  password: 'secret',
+};
+
+const ANTONINAS_NHOOD_ID = 2;
+const RADUS_REQUEST_ID = 2;
+const ANTONINAS_RESPONSE_ID = 1;
+const ANTONINAS_USER_ID = 2;
+
+const loginUser = async (loginData: LoginData) => {
+  const loginResponse = await api.post('/api/login').send(loginData);
+
+  return loginResponse;
+};
+
+beforeAll(async () => {
+  await seed();
+});
+
+// We have ANTONINAS_RESPONSE (Antonina's user_id = 2, response_id = 1)
+// The response is associated with RADUS_REQUEST (request_id = 2
+// created by the user RADU, user_id = 4)
+// RADUS_REQUEST is in ANTONINAS_NEIGHBORHOOD (neighborhood_id = 2, admin_id = 2)
+
+describe('Tests for updating a request: PUT /responses/:id', () => {
+  let token: string;
+
+  beforeAll(async () => {
+    const loginResponse: Response = await loginUser(ANTONINAS_LOGIN_DATA);
+    token = loginResponse.body.token;
+  });
+
+  beforeEach(async () => {
+    await seed();
+  });
+
+  test("User cannot update httpResponse if they aren't logged in", async () => {
+    const newData: UpdateResponseData = { content: 'new content' };
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .send(newData);
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.status).toBe(401);
+    expect(httpResponse.body.error).toBe('user not signed in');
+    expect(responseInDbAfterUpdate.content).not.toBe(newData.content);
+  });
+
+  test("User cannot update httpResponse if they didn't create it", async () => {
+    const loginResponse: Response = await loginUser(BOBS_LOGIN_DATA);
+    const bobToken: string = loginResponse.body.token;
+
+    const newData: UpdateResponseData = { content: 'test content' };
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${bobToken}`)
+      .send(newData);
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.status).toBe(401);
+    expect(httpResponse.body.error).toBe('User does not have edit rights.');
+    expect(responseInDbAfterUpdate.content).not.toBe(newData.content);
+  });
+
+  test('Update with invalid properties raises an error', async () => {
+    const invalidData = { invalid: 'Test' };
+
+    const httpResponse: Response = await api
+      .put(`/api/responses/${ANTONINAS_RESPONSE_ID}`)
+      .set('Authorization', `Bearer ${token}`)
+      .send(invalidData);
+
+    const responseInDbAfterUpdate = await testHelpers.getSingleResponse(
+      ANTONINAS_RESPONSE_ID,
+    );
+
+    expect(httpResponse.status).toBe(400);
+    // Does the following error message makes sense here?
+    // Probably a very minor thing which can be ignored
```

Comment:
We could make it more generic, like "Invalid response property" or sth

Reply:
Ignoring for now.